iree/turbine/kernel/wave/wave.py

```diff
@@ -488,13 +539,24 @@ def _trace_and_get_kernel_signature(
         dispatch_codegen.StreamExecutable,
         kernel_codegen.KernelSignature,
         str,
+        KernelLaunchInfo,
     ]:
         # Issue a warning if IREE ver is too low.
         # Warning will only be issued if we are compiling the kernel and won't
         # if we are using cached kernel as we don't want to add any additional
         # overhead to 'happy' path.
         _warn_iree_is_too_old()
 
+        # Store kernel launch info for the wave runtime.
+        kernel_launch_info = KernelLaunchInfo()
+
+        # Build wave runtime, if specified.
+        run_config = kwargs.get("run_config", {})
+        if run_config.get("wave_runtime", None):
+            # Remove any existing hsaco files in this directory.
+            remove_files_with_extension(WAVE_RUNTIME_DIR, ".hsaco")
```

Comment:
wouldn't this mean between python runs, we are re-compiling the kernels?

Reply:
We do this because we want to make sure that at any point of time there's only a single hsaco file located inside the wave runtime directory inside the cache directory. This only happens in situations where we are not caching any kernels and so it is okay to remove the last compiled kernel.

Reply:
Is this because we would use `(CACHE_BASE_DIR / kernel_hash / kernel_hash) + ".hsaco"` if we have cached and `WAVE_RUNTIME_DIR/*.hsaco` is only used as a temporary location/cases where we do not have it cached?

Reply:
yes thats right.

Reply:
Doesn't have to be this PR, but we should add this in a comment somewhere since it is not very obvious (at least for me :D)

Reply:
Sounds good, will add in next PR.

Reply:
Runners dont seem to be picking this up, so will add the comment in this PR and resubmit.